Documentation/git-diff.txt

```diff
@@ -77,8 +77,16 @@ two blob objects, or changes between two files on disk.
 
 Just in case you are doing something exotic, it should be
 noted that all of the <commit> in the above description, except
-in the last two forms that use ".." notations, can be any
-<tree>.
+in the last form, that use the "\..." notation, can be any
+<tree>. For instance, if you want to make diffs against an empty
+tree, you can create a tag pointing to the empty tree:
```

Comment:
I'd stop at fixing the original that implied ".." cannot be used to say "..." cannot, and would not add the empty-tree example at all.

While comparing with an empty tree and making it handy to do so with a tag is not technically incorrect per-se, checking a working tree for whitespace breakages by comparing everything with an empty tree is something we definitely do not want to encourage.

Reply:
Well, what form is use to create empty tree is just a matter of taste. I also more like the first form.

What about the empty tree example. I added this for two reason. First, I can't think of another reason to use a tree-ish with git diff. Second, the diff with empty tree is a real problem on what some people looking for solution  (for various reasons):
https://stackoverflow.com/questions/14564034/creating-a-git-diff-from-nothing/54946189#54946189
https://stackoverflow.com/questions/9765453/is-gits-semi-secret-empty-tree-object-reliable-and-why-is-there-not-a-symbolic/54945479#54945479
And the solutions are not simple nor obvious. So I think it will be helpful to mention it.

But well, if you get it only without empty tree example, I'll agree. What will be better: create new pull request or rebase this?

Reply:
> What will be better: create new pull request or rebase this?

First of all, Git's contributions are not accepted via PRs (instead it wants the patches to be sent to the Git mailing list)... but there is a way to use PRs to generate these patches.

So what you want to do is to rebase this to `master`, adjust according to the reviewers comments (if you disagree with them, you will have to find ways to convince them, preferably in the form of good commit messages).

It would be furthermore a good idea to change the format of the first line of the commit message to the suggested format (see https://git-scm.com/docs/SubmittingPatches#describe-changes): start with an area prefix (in your case, probably "user manual: "), keep it to 50 characters if possible, use the imperative form and do not use a full stop at the end of the first line.

Please also make sure that your commit message focuses on answering the question "why?" more than on "how?", and that it wraps at <= 76 columns per line. Also do make sure to add your [sign off](https://git-scm.com/docs/SubmittingPatches#dco).

Finally send the patch to the mailing list for review. You can use [GitGitGadget](https://gitgitgadget.github.io), [submitGit](https://submitgit.herokuapp.com) or [send it manually](https://github.com/git-for-windows/git/blob/master/CONTRIBUTING.md#submit-your-patch).

Reply:
> It would be furthermore a good idea to change the format of the first line of the commit message to the suggested format (see https://git-scm.com/docs/SubmittingPatches#describe-changes): start with an area prefix (in your case, probably "user manual: "), keep it to 50 characters if possible, use the imperative form and do not use a full stop at the end of the first line.
> 
> Please also make sure that your commit message focuses on answering the question "why?" more than on "how?", and that it wraps at <= 76 columns per line. Also do make sure to add your [sign off](https://git-scm.com/docs/SubmittingPatches#dco).

I can't undestand. The patch is just path, here it is:
```
--- a/Documentation/git-diff.txt
+++ b/Documentation/git-diff.txt
@@ -77,8 +77,16 @@ two blob objects, or changes between two files on disk.

 Just in case you are doing something exotic, it should be
 noted that all of the <commit> in the above description, except
-in the last two forms that use ".." notations, can be any
-<tree>.
+in the last form, that use the "\..." notation, can be any
+<tree>. For instance, if you want to make diffs against an empty
+tree, you can create a tag pointing to the empty tree:
+
+'git tag' empty $('git hash-object' -t tree /dev/null)
+
+And use it for the 'git diff', for instance, to check a working tree
+for whitespaces:
+
+'git diff' --check empty

 For a more complete list of ways to spell <commit>, see
 "SPECIFYING REVISIONS" section in linkgit:gitrevisions[7].
```
Where is here a commit message?